File: platform-core-service/platform-core-service.Common/Models/DTOs/EntityDTO/Moderation/AdminQueueEntryDTO.cs
using platform_core_service.Common.Entities.BaseEntity;
using platform_core_service.Common.Models.DTOs.EntityDTO.Post;
using platform_core_service.Common.Utils.Enums;

namespace platform_core_service.Common.Models.DTOs.EntityDTO.Moderation
{
    public class AdminQueueEntryDTO : IBaseKey<string>
    {
        public string Id { get; set; } = null!;
        public ModerationTargetType TargetType { get; set; }
        public string TargetId { get; set; } = null!;
        public string PostTitle { get; set; } = null!;
        public string PostContent { get; set; } = null!;
        public string AuthorId { get; set; } = null!;
        public SelectPostAuthorDTO Author { get; set; }
        public string EntityType { get; set; }
        public string Reason { get; set; } = null!;
        public float Tier1Score { get; set; }
        public string Tier2Reasoning { get; set; } = null!;

        public string? AssignedModeratorId { get; set; }
        public DateTimeOffset? ResolvedAt { get; set; }
        public string? Resolution { get; set; }
        public string? ModeratorNote { get; set; }

        public DateTimeOffset CreatedAt { get; set; }
    }
}

File: frontend/src/types/admin/admin-queue-entry-dto.ts
import { SelectPostAuthorDTO } from './admin-post-dto';

export interface AdminQueueEntryDTO {
  id: string;
  targetType: 'Post' | 'Answer' | 'Comment' | 0 | 1 | 2;
  targetId: string;
  postId?: string;
  postTitle: string;
  postContent: string;
  authorId: string;
  author: SelectPostAuthorDTO;
  entityType: string;
  reason: string;
  tier1Score: number;
  tier2Reasoning: string;
  assignedModeratorId?: string;
  resolvedAt?: string;
  resolution?: string;
  createdAt: string;
}

export interface AdminQueueResolveDTO {
  id: string;
  resolution: 'Approved' | 'Rejected';
  moderatorNote?: string;
}
